### PR TITLE
Personalization for PrestaShop v2.3.0

### PR DESCRIPTION
**Pull request by kryysler**
* Added order status info to the order confirmation API requests
* Added new product update API instead of deprecated re-crawl feature
* Added possibility to add a product to the shopping cart directly from the recommendations using JS
* Add additional checks to the manufacturer and category pages that we actually can find the model
* Refactor Nosto model validation so we can allow incomplete data in the tagging

### CODE REVIEW

File: classes/helpers/product-operation.php [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+	 * The currency will be the first found for the shop, but it defaults to the PS default currency
+	 * if no shop specific one is found.
+	 *
+	 * @param int $id_lang the language ID to add to the new context.
+	 * @param int $id_shop the shop ID to add to the new context.
+	 * @return Context the new context.
+	 */
+	protected function makeContext($id_lang, $id_shop)
+	{
+		if (_PS_VERSION_ >= '1.5')
```
Comment (kryysler) [inline classes/helpers/product-operation.php]: This method is a bit hacky to get the product page url and image url to have the correct domain, path and "friendly url" settings. We run this per shop, so we need to replace the context for each shop.

**Comment by mridang** [comment]
@Quetzacoalt91 Hi Tomas, have you had any luck with this? @kryysler and me would like to get this pull-request out.

**Comment by gaillafr** [comment]
The error message is misdisplayed : https://www.dropbox.com/s/qhia8ojqhnzs3eq/Screenshot%202015-06-08%2011.28.09.png?dl=0

**Comment by gaillafr** [comment]
The configuration form iFrame took me almost 30s to load (too long for our merchants).

**Comment by kryysler** [comment]
We are looking into these issues and will get back to you asap.

**Comment by kryysler** [comment]
Fixed the message encoding issue and improved the response times of the dashboard page in the iframe

**Comment by kryysler** [comment]
Do you have all you need to move forward with this, or can we provide anything else to get this PR accepted?

**Comment by gaillafr** [comment]
Hi @kryysler, when I try to merge your PR I'm facing an error on my shop:
```
Fatal error: Interface 'NostoValidatableInterface' not found in /home/gaillafr/www/public_html/prestashop/modules/nostotagging/classes/models/order.php on line 30
[PrestaShop] Fatal error in module file :/home/gaillafr/www/public_html/prestashop/modules/nostotagging/classes/models/order.php:
Interface 'NostoValidatableInterface' not found
```

Please, let me know if you can reproduce that.

Thank you,
Kind regards.

**Comment by kryysler** [comment]
I cannot reproduce it. That interface is available in the SDK that is a sub-repo (https://github.com/Nosto/php-sdk/blob/d07226eec659422d88a191708c7046a5594e57ef/src/interfaces/NostoValidatableInterface.php#L39).

The sub-repo reference is also correct in this PR.

**Comment by gaillafr** [comment]
Sorry @kryysler, I've tried to merged it again (new clone, etc.) locally but the problem still occurs :-/

**Comment by kryysler** [comment]
@gaillafr could you check that the sub-module in libs/nosto/php-sdk is updated after the merge (should be changeset d07226e)? And that the file exists ("libs/nosto/php-sdk/src/interfaces/NostoValidatableInterface.php") ?

**Comment by gaillafr** [comment]
Yes, I've tried to update the two submodules but it didn't change anything.

**Comment by kryysler** [comment]
Can you check that the interface is loaded in the submodule config.inc.php file. This line:
https://github.com/Nosto/php-sdk/blob/d07226eec659422d88a191708c7046a5594e57ef/src/config.inc.php#L51

**Pull request merged**